dartsim/src/EntityManagementFeatures.cc

```diff
@@ -436,25 +442,41 @@ Identity EntityManagementFeatures::GetJoint(
 Identity EntityManagementFeatures::GetJoint(
     const Identity &_modelID, const std::string &_jointName) const
 {
-  DartJoint *const joint =
-      this->ReferenceInterface<ModelInfo>(_modelID)->model->getJoint(
-          _jointName);
+  const auto &modelInfo = this->ReferenceInterface<ModelInfo>(_modelID);
 
-  // If the joint doesn't exist in "joints", it means the containing entity has
-  // been removed.
-  if (this->joints.HasEntity(joint))
+  auto worldID = this->GetWorldOfModelImpl(_modelID);
+  if (worldID == INVALID_ENTITY_ID)
   {
-    const std::size_t jointID = this->joints.IdentityOf(joint);
-    return this->GenerateIdentity(jointID, this->joints.at(jointID));
+    gzerr << "World of model [" << modelInfo->model->getName()
+          << "] could not be found for joint [" << _jointName
+          << "]\n";
+    return this->GenerateInvalidId();
   }
-  else
+
+  auto world = this->worlds.at(worldID);
+  const std::string fullJointName = ::sdf::JoinName(
+      world->getName(),
+      ::sdf::JoinName(modelInfo->model->getName(), _jointName));
```

Comment:
Can this use `FullyScopedJointName` instead?

Reply:
Good spot - updated in 47b46868